### PR TITLE
PR: Improve release instructions

### PR DESCRIPTION
**Pull request by ccordoba12**
- This streamlines and simplifies our release instructions.
- It also fixes some errors in our Changelog and Announcements files.

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -5,31 +5,85 @@ To release a new version of Spyder you need to follow these steps:
...
+  - `setup.py`
+  - `spyder/dependencies.py`
+  - `spyder/plugins/ipythonconsole/plugin.py`
+
+* Commit with
+
+      git add .
+      git commit -m "Update core dependencies"
+
+* Update their respective subrepos with the follwing commands, but only if new versions are available!
```
Comment (dalthviz) [inline RELEASE.md:83]:
```suggestion
* Update their respective subrepos with the following commands, but only if new versions are available!
```

File: RELEASE.md [hunk 2]
```diff
@@ -42,21 +96,21 @@ To release a new version of Spyder you need to follow these steps:
...
 
-* git commit with "Update Announcements".
+* `git commit -m "Update Announcements"`
 
 * `git clean -xfdi` and select option `1`.
 
 * Update version in `__init__.py` (set release version, remove 'dev0')
 
-* `git add .` and git commit with `Release X.X.X`
+* `git add .` and `git commit .m "Release X.X.X"`
```
Comment (dalthviz) [inline RELEASE.md:109]:
```suggestion
* `git add .` and `git commit -m "Release X.X.X"`
```

**Review by dalthviz — COMMENTED** [review]
Thanks @ccordoba12 Left some comments with suggestions. Besides that this LGTM :+1:

**Review by dalthviz — COMMENTED** [review]
@ccordoba12 checking today's notifications, a PR to update the `conda-forge` package for `spyder-kernels` was opened, should we add here some instructions about that (merging `conda-forge` related PRs after a release)?

**Comment by ccordoba12** [comment]
Good point! I'll add a new `After the release` section to mention those details.

**Review by dalthviz — APPROVED** [review]
Thanks @ccordoba12 LGTM :+1:

**Pull request merged**